management-api-agent-common/src/main/java/com/datastax/mgmtapi/NodeOpsProvider.java

```diff
@@ -732,31 +735,67 @@ public void clearSnapshots(
     }
   }
 
+  // This version of repair takes beginToken and endToken, which does not allow for non-contiguous ranges to be repaired under
+  // a single run.
   @Rpc(name = "repair")
   public void repair(
       @RpcParam(name = "keyspaceName") String keyspace,
       @RpcParam(name = "tables") List<String> tables,
-      @RpcParam(name = "full") Boolean full)
+      @RpcParam(name = "full") Boolean full,
+      @RpcParam(name = "beginToken") BigInteger beginToken,
+      @RpcParam(name = "endToken") BigInteger endToken,
+      @RpcParam(name = "repairParallelism") RepairParallelism repairParallelism,
+      @RpcParam(name = "datacenters") Collection<String> datacenters,
+      @RpcParam(name = "repairThreadCount") int repairThreadCount)
       throws IOException {
     // At least one keyspace is required
-    if (keyspace != null) {
-      // create the repair spec
-      Map<String, String> repairSpec = new HashMap<>();
-
-      // add any specified tables to the repair spec
-      if (tables != null && !tables.isEmpty()) {
-        // set the tables/column families
-        repairSpec.put(RepairOption.COLUMNFAMILIES_KEY, String.join(",", tables));
-      }
+    assert(keyspace != null);
+    Map<String, String> options = new HashMap<>();
+    options.put(RepairOption.PARALLELISM_KEY, repairParallelism.getName());
+    options.put(RepairOption.INCREMENTAL_KEY, Boolean.toString(!full));
+    options.put(
+        RepairOption.JOB_THREADS_KEY,
+        Integer.toString(repairThreadCount == 0 ? 1 : repairThreadCount));
+    options.put(RepairOption.TRACE_KEY, Boolean.toString(Boolean.FALSE));
+    options.put(RepairOption.COLUMNFAMILIES_KEY, StringUtils.join(tables, ","));
+    if (full) {
+      options.put(
+          RepairOption.RANGES_KEY, StringUtils.join(beginToken + ":" + endToken)
+      );
+    }
 
-      // handle incremental vs full
-      boolean isIncremental = Boolean.FALSE.equals(full);
-      repairSpec.put(RepairOption.INCREMENTAL_KEY, Boolean.toString(isIncremental));
-      if (isIncremental) {
-        // incremental repairs will fail if parallelism is not set
-        repairSpec.put(RepairOption.PARALLELISM_KEY, RepairParallelism.PARALLEL.getName());
-      }
-      ShimLoader.instance.get().getStorageService().repairAsync(keyspace, repairSpec);
+  }
+
+  // This version of repair() takes a list of RingRange objects instead of begin/end tokens.
+  @Rpc(name = "repair")
```

Comment:
This won't work. We'll need a different Rpc `name` value or the registration will complain of duplicate names.